### PR TITLE
Add pre-install script to copy env before install

### PR DESCRIPTION
**Pull request by ryanwarsaw**
This verifies that the .env file has been copied before trying to run the post-install script.

### CODE REVIEW

File: scripts/ensure-env.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+var fs = require("fs");
+var env = require("path").join(__dirname, "../", ".env");
```
Comment (Pomax) [inline scripts/ensure-env.js]: `join(__dirname, "..", ".env");` (path takes care of putting the correct delimiter in between the various path segments)

File: scripts/ensure-env.js [hunk 2]
```diff
@@ -0,0 +1,7 @@
...
+
+fs.exists(env, function (exists) {
```
Comment (Pomax) [inline scripts/ensure-env.js]: We don't need async here, so keeping things synchronous actually yields a more predictable code path:
```
var exists = fs.existsSync(env);
if (exists) ...
...
```

File: README.md [hunk 3]
```diff
@@ -17,7 +17,6 @@
 1. Fork and clone this repository
 2. Navigate to the directory of the repository, e.g. `cd id.webmaker.org`
 3. `npm install` to install dependencies
-4. Set configuration variables, by default `cp sample.env .env` or `copy sample.env .env` on Windows
```
Comment (Pomax) [inline README.md]: ❤️ :

**Comment by ryanwarsaw** [comment]
cc: @gideonthomas @cadecairos R+?

**Comment by Pomax** [comment]
R+ on my side

**Comment by gideonthomas** [comment]
I'm pretty sure the `pre-install` script is run on heroku as well. Won't this then use what's in sample.env over what's in the actual env?

**Comment by Pomax** [comment]
Easy enough to fix: this is a node script, so we can simply add an `if (process.env.NODE === "PRODUCTION") process.exit(0);`

File: scripts/ensure-env.js [hunk 4]
```diff
@@ -0,0 +1,5 @@
+var fs = require("fs");
```
Comment (Pomax) [inline scripts/ensure-env.js]: based on https://github.com/mozilla/id.webmaker.org/pull/428#issuecomment-279095560 we'll need a check here to make sure we're on a local dev's machine, not on heroku...

**Comment by ryanwarsaw** [comment]
@Pomax @gideonthomas Thanks for catching that, i went ahead and made the review changes requested, and learned a lot about Node's process API and environment-related stuff 👀  R+?

I also wanted to pick your brains on a question related to styling and JS, I'm used to working with languages like Java where there are several prominent standards (Google & Oracle's guides). However, with JS it seems to be much more diverse and spread out across the board. Is there a prominent standard with JS, or is it one of those things where as long as it passes lint it's okay?

🤔 💭

**Comment by ryanwarsaw** [comment]
I just realized I have to check for ```staging``` as well, I'm going to do it in the morning since it's 1 AM and I can't think straight. 😴

**Comment by Pomax** [comment]
@ryanwarsaw you can just check for a `NODE_ENV` var at all. If there is one, don't do anything, because whoever has that set clearly has special intentions.

**Comment by ryanwarsaw** [comment]
@Pomax R+?

File: scripts/ensure-env.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 var fs = require("fs");
 var env = require("path").join(__dirname, "..", ".env");
 
-if (fs.existsSync(env) || process.env.NODE_ENV == "production") process.exit(0);
+if (fs.existsSync(env) || !(typeof process.env.NODE_ENV === 'undefined')) process.exit(0);
```
Comment (Pomax) [inline scripts/ensure-env.js]: small nit: ` !(typeof process.env.NODE_ENV === 'undefined')` => `typeof process.env.NODE_ENV !== 'undefined'`, just to remove some parentheses.

File: scripts/ensure-env.js [hunk 6]
```diff
@@ -0,0 +1,5 @@
+var fs = require("fs");
+var env = require("path").join(__dirname, "..", ".env");
+
+if (fs.existsSync(env) || !(typeof process.env.NODE_ENV === 'undefined')) process.exit(0);
```
Comment (Pomax) [inline scripts/ensure-env.js]: small nit: `!(typeof process.env.NODE_ENV === 'undefined')` => `typeof process.env.NODE_ENV !== 'undefined'`

**Comment by ryanwarsaw** [comment]
@Pomax R+? Thanks for the reviews.

**Comment by Pomax** [comment]
yup R+

**Pull request merged**